src/swayncDaemon/swayncDaemon.vala

```diff
@@ -258,6 +258,12 @@ namespace SwayNotificationCenter {
             noti_daemon.control_center.close_notification (id, true);
         }
 
+        /** Activates the `noti_id` action of the latest notification */
+        public void invoke_action (uint32 noti_id)
```

Comment:
invoke_action is a bit misleading, maybe latest_invoke_action? Also change noti_id to action_index or something similar

Same in notidaemon.vala